tsconfig.typecheck.json

```diff
@@ -2,8 +2,8 @@
 	"extends": "./tsconfig.base.json",
 	"compilerOptions": {
 		"noEmit": true,
-		"lib": ["DOM", "ESNext"],
-		"types": ["vitest/globals"]
+		"types": ["vitest/globals"],
+		"lib": ["dom", "esnext"]
```

Comment:
Unnecessary? lol

Reply:
ik i'll revert it. It was bcz I was testing `happy-dom` and I added `"lib": ["dom", "esnext"]` in one `tsconfig` and after removing that deps, I tought I added `lib` here too so I removed it but then saw tests are failing bcz it was present earlier so added back lol

Reply:
Entire file is deleted in favour of having the few required compiler options in `tsconfig.eslint.json`. They don't affect eslint in any way and this saves on a file in the root folder.